bundles/org.openhab.binding.teleinfo/src/main/resources/ESH-INF/thing/common-cbemm-channel-types.xml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="teleinfo"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<channel-type id="imaxType">
+		<item-type>Number:ElectricCurrent</item-type>
+		<label>IMAX</label>
+		<description>IMAX</description>
+		<category>Energy</category>
+		<state pattern="%d A" readOnly="true" />
+	</channel-type>
+
+	<channel-type id="iinstType">
+		<item-type>Number:ElectricCurrent</item-type>
+		<label>IINST</label>
+		<description>IINST</description>
+		<category>Energy</category>
+		<state pattern="%d A" readOnly="true" />
+	</channel-type>
+
+	<channel-type id="currentPowerType">
+		<item-type>Number:Power</item-type>
+		<label>currentpower</label>
+		<description>Instantaneous power</description>
+		<category>Energy</category>
+		<state pattern="%d W" readOnly="true" />
+		<config-description>
+			<parameter name="powerFactor" type="integer">
+				<label>Power factor</label>
```

Comment:
Actually "power factor" describes the phase shift in AC systems. Are you sure this is a good wording?

Reply:
I'm not an electricity expert. My work is based on this blog: http://www.magdiblog.fr/gpio/teleinfo-edf-suivi-conso-de-votre-compteur-electrique/
This blog references the "Facteur de puissance" in Wikipedia: https://fr.wikipedia.org/wiki/Facteur_de_puissance (https://en.wikipedia.org/wiki/Power_factor in English)

Reply:
I think a good wording could be "Nominal Voltage". Power factor is the cosinus of phase shift:  a real number between 0 and 1. The technical name is "rms (root mean square) voltage" but perhaps it is too technical. I will make the change in "Nominal Voltage" if it suits everybody.

Reply:
I think that suits better.

Reply:
You seem more expert than me about this subject :-) Ok for me